src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/ThemeManager.cs

```diff
@@ -0,0 +1,216 @@
+using Standard;
+using System.Windows.Appearance;
+using System.Windows.Media;
+using Microsoft.Win32;
+using System.Collections;
+using System.Collections.Generic;
+using System.Windows.Interop;
+
+namespace System.Windows;
+
+internal static class ThemeManager
+{
+
+    #region Constructor
+
+    static ThemeManager()
+    {
+        _cachedThemeDictionaryUris = new List<Uri>();
+
+        if (Application.Current != null)
+        {
+            foreach (ResourceDictionary mergedDictionary in Application.Current.Resources.MergedDictionaries)
+            {
+                if (mergedDictionary.Source != null && mergedDictionary.Source.ToString().EndsWith("FluentWindows.xaml"))
+                {
+                    _isFluentWindowsThemeEnabled = true;
+                    break;
+                }
+            }
+        }
+
+        if(_isFluentWindowsThemeEnabled)
+        {
+            _currentApplicationTheme = GetSystemTheme();
+            DwmColorization.UpdateAccentColors();
+        }
+    }
+
+    #endregion
+
+    #region Internal Methods
+
+    internal static void ApplySystemTheme(bool forceUpdate = false)
+    {
+        string systemTheme = GetSystemTheme();
+        bool useLightMode = GetUseLightTheme();
+        Color systemAccentColor = DwmColorization.GetSystemAccentColor();
+
+        var windows = Application.Current?.Windows;
+        if(windows != null)
```

Comment:
If there is no Application, none of the code in the ApplyTheme will not run. Is that alright?

Reply:
Actually, if there is no application then these methods won't be called. The flag for IsFluentThemesEnabled, is set to true only when we have an instance of application.

Moreover, in the current theme manangement, we add the resources in App.xaml, and we update the resources by accessing Applicaiton.Current.Resources. So, without Application none of this works.